homeassistant/components/samsungtv/bridge.py

```diff
@@ -566,17 +566,25 @@ async def async_device_info(self, force: bool = False) -> dict[str, Any] | None:
         """Try to gather infos of this TV."""
         if self._rest_api is None:
             assert self.port
-            rest_api = SamsungTVAsyncRest(
+            self._rest_api = SamsungTVAsyncRest(
                 host=self.host,
                 session=async_get_clientsession(self.hass),
                 port=self.port,
                 timeout=TIMEOUT_WEBSOCKET,
             )
 
-        with contextlib.suppress(*REST_EXCEPTIONS):
-            device_info: dict[str, Any] = await rest_api.rest_device_info()
```

Comment:
With `contextlib.suppress` it's hard to trace the various attempts at discovering the correct connection mode